server/utils/passport/local-signup.js

```diff
@@ -1,91 +1,63 @@
-      if (err) {
-        return res.redirect(`${basePath}/login?e=${err}`)
-      }
-      if (user) {
-        return res.redirect(`${basePath}/signup?e=existingUser`)
-      }
-      const password = reqBody.password
-      const username = reqBody.username
-      const email = reqBody.email
-      const display_name = reqBody.display_name
-      const { appDb } = req.app.locals
-      const hashedPW = hash(password)
-      appDb.collection(collections.users).findOneAndUpdate(
-        { uid: username },
-        {
-          $set: {
-            display_name: display_name,
-            title: '',
-            department: '',
-            company: '',
-            mail: email,
-            member_of: '',
-            bad_pwd_count: 0,
-            lockout_time: 0,
-            last_logoff: Date.now(),
-            last_logon: Date.now(),
-            account_expires: null,
+    async function (err, user, reqBody) {
+      try {
+        if (err) return res.redirect(`${basePath}/login?e=${err}`)
+
+        if (user) return res.redirect(`${basePath}/signup?e=existingUser`)
+
+        const password = reqBody.password
+        const uid = reqBody.username
+        const email = reqBody.email
+        const display_name = reqBody.display_name
+        const { appDb } = req.app.locals
+        const hashedPW = hash(password)
+        const account_expires = moment().add(2, 'years').format()
+
+        await appDb.collection(collections.users).insertOne({
+          uid,
+          ...SIGN_UP_UPDATE_DATA({
+            display_name,
             password: hashedPW,
-            ldap: false,
-            force_reset_pw: false,
+            mail: email,
             realms: config.app.realms,
-            icon: '',
-            access: [],
-            blocked: false,
-            role: 'member',
-          },
-        },
-        {
-          projection: {
-            _id: 0,
-            uid: 1,
-            display_name: 1,
-            acl: 1,
-            role: 1,
-            icon: 1,
-            mail: 1,
-          },
-          returnOriginal: false,
-          upsert: true,
-        },
-        function (err, userinfo) {
-          if (err) {
-            console.error(err.message)
-          }
+            account_expires,
+          }),
+        })
+
+        await appDb
+          .collection(collections.configs)
+          .insertOne(DEFAULT_CONFIG_AT_SIGNUP(uid))
 
-          var defaultConfig = {
-            owner: username,
-            config: defaultUserConfig,
-            name: 'default',
-            type: 'matrix',
-            readers: [username],
-            created: new Date().toUTCString(),
-          }
-          appDb.collection(collections.configs).insertOne(defaultConfig)
+        req.session.role = INITIAL_ROLE
+        req.session.uid = uid
+        req.session.display_name = display_name
+        req.session.mail = email
+        req.session.celery_tasks = []
+        req.session.icon = ''
```

Comment:
do we need to store this info in the session or can we get it from the user?

Reply:
good question I don't think we need it at sign up. We don't give the user access right away

Reply:
do we ever need to store more than the `uid` in the session? If we only had the uid in the session then we could get the user from the database to check the other attributes. Not sure if that makes sense or not.